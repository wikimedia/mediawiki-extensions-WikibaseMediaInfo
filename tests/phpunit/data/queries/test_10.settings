{
	"title": "Query plain equivalent when stemming is not available",
	"term": "yet another search term",
	"entityIdsForTerm": {
		"yet another search term": []
	},
	"userLanguage": "de",
	"fallbackLangs": [ "fr" ],
	"stemmingSettings": {
		"de": {
			"query": false
		},
		"fr": {
			"query": true
		}
	},
	"settings": {
		"boost": {
			"statement": {"P999": 0},
			"descriptions.$language": 2,
			"descriptions.$language.plain": 0,
			"category": 0,
			"category.plain": 0,
			"title": 0,
			"title.plain": 0,
			"redirect.title": 0,
			"redirect.title.plain": 0,
			"text": 0,
			"text.plain": 0,
			"auxiliary_text": 0,
			"auxiliary_text.plain": 0,
			"file_text": 0,
			"file_text.plain": 0,
			"heading": 0,
			"heading.plain": 0,
			"suggest": 0
		}
	}
}
