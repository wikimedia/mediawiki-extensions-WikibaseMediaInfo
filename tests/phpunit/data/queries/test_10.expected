{
    "bool": {
        "must": [
            {
                "function_score": {
                    "query": {
                        "bool": {
                            "minimum_should_match": 1,
                            "should": [
                                {
                                    "dis_max": {
                                        "tie_breaker": 1,
                                        "queries": [
                                            {
                                                "bool": {
                                                    "minimum_should_match": 0,
                                                    "filter": [
                                                        {
                                                            "multi_match": {
                                                                "query": "yet another search term",
                                                                "fields": [
                                                                    "all",
                                                                    "all.plain"
                                                                ],
                                                                "operator": "and"
                                                            }
                                                        }
                                                    ],
                                                    "should": [
                                                        {
                                                            "match": {
                                                                "descriptions.de.plain": {
                                                                    "query": "yet another search term",
                                                                    "boost": 2
                                                                }
                                                            }
                                                        },
                                                        {
                                                            "match": {
                                                                "descriptions.fr": {
                                                                    "query": "yet another search term",
                                                                    "boost": 2
                                                                }
                                                            }
                                                        }
                                                    ]
                                                }
                                            }
                                        ]
                                    }
                                },
                                {
                                    "match_none": {}
                                }
                            ]
                        }
                    },
                    "functions": [
                        {
                            "script_score": {
                                "script": {
                                    "source": "100 \/ ( 1 + exp( -1 * ( _score + intercept ) ) )",
                                    "params": {
                                        "intercept": 1
                                    },
                                    "lang": "expression"
                                }
                            }
                        }
                    ]
                }
            }
        ],
        "filter": [
            {
                "terms": {
                    "namespace": [
                        0
                    ]
                }
            }
        ]
    }
}